plugin/src/main/java/net/aufdemrand/denizen/objects/dLocation.java

```diff
@@ -2419,6 +2433,28 @@ else if (getBlock().getType() == Material.FLOWER_POT) {
             }
         }
 
+        // <--[mechanism]
+        // @object dLocation
+        // @name custom_name
+        // @input Element
+        // @description
+        // Sets the custom name of the block.
+        // Use no value to reset the block's name.
+        // @tags
+        // <l@location.custom_name>
+        // -->
+        if (mechanism.matches("custom_name")) {
+            if (getBlock().getState() instanceof Nameable) {
+                String title = mechanism.getValue().asString();
+                if (!mechanism.hasValue()) {
+                    title = null;
+                }
```

Comment:
Yeah ^ that is... janky. It technically works but _yikes_
```
getValue().operateOnValue();
if (getValue() == null) {
  // woops!
```

Technically getValue isn't null when there's no value but you should code as though it is